packages/desktop-client/src/components/reports/graphs/SpendingGraph.tsx

```diff
@@ -129,7 +137,15 @@ export function SpendingGraph({
   const balanceTypeOp = 'cumulative';
   const thisMonth = monthUtils.currentMonth();
   const lastMonth = monthUtils.subMonths(monthUtils.currentMonth(), 1);
-  const selection = mode.toLowerCase() === 'average' ? 'average' : lastMonth;
+  const lastYear = monthUtils.prevYear(monthUtils.currentMonth());
+  let selection;
+  if (mode.toLowerCase() === 'average') {
+    selection = 'average';
+  } else if (mode.toLowerCase() === 'last month') {
+    selection = lastMonth;
+  } else {
+    selection = lastYear;
+  }
```

Comment:
Generally should try to avoid nested if statements. A switch statement would work better here.